samples/openthread/libraries/prj.conf

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+CONFIG_NEWLIB_LIBC=y
+CONFIG_ENTROPY_CC310=n
+
+# Generic networking options
+CONFIG_NETWORKING=y
+CONFIG_NET_UDP=y
+CONFIG_NET_IPV6=y
+CONFIG_NET_CONNECTION_MANAGER=y
+
+# Kernel options
+CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_ENTROPY_GENERATOR=y
+CONFIG_INIT_STACKS=y
+
+# Logging
+CONFIG_NET_LOG=y
+CONFIG_LOG=y
+CONFIG_NET_STATISTICS=y
+CONFIG_PRINTK=y
+CONFIG_OT_COMMAND_LINE_INTERFACE_LOG_LEVEL_DBG=y
+CONFIG_NET_CONFIG_LOG_LEVEL_INF=y
+
+# Network shell
+CONFIG_NET_SHELL=y
+
+CONFIG_SHELL_ARGC_MAX=26
+CONFIG_SHELL_CMD_BUFF_SIZE=416
+
+CONFIG_CPLUSPLUS=y
+
+CONFIG_NET_CONFIG_SETTINGS=y
+CONFIG_NET_IPV6_NBR_CACHE=n
+CONFIG_NET_IPV6_MLD=n
+CONFIG_NET_CONFIG_NEED_IPV6=y
+CONFIG_NET_CONFIG_INIT_TIMEOUT=0
+
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
+CONFIG_NET_TX_STACK_SIZE=2048
+CONFIG_SHELL_STACK_SIZE=3072
+
+CONFIG_NET_L2_OPENTHREAD=y
+
+CONFIG_OPENTHREAD_DEBUG=y
+CONFIG_OPENTHREAD_L2_DEBUG=y
+CONFIG_OPENTHREAD_L2_LOG_LEVEL_INF=y
+
+CONFIG_OPENTHREAD_CHANNEL=11
+CONFIG_OPENTHREAD_THREAD_PREEMPTIVE=y
+
+# Enable joiner and commissioner optional features
+CONFIG_OPENTHREAD_COMMISSIONER=y
+CONFIG_OPENTHREAD_JOINER=y
+CONFIG_MBEDTLS_HEAP_SIZE=10240
+
+# Other optional OpenThread features
+CONFIG_OPENTHREAD_DHCP6_CLIENT=y
+CONFIG_OPENTHREAD_SLAAC=y
+CONFIG_OPENTHREAD_BORDER_ROUTER=y
+
+# Thread by default registers quite a lot addresses.
+CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT=6
+CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=8
+
+# Other OpenThread dependencies
+CONFIG_FLASH=y
+CONFIG_FLASH_PAGE_LAYOUT=y
+CONFIG_FLASH_MAP=y
+CONFIG_MPU_ALLOW_FLASH_WRITE=y
+CONFIG_NVS=y
+CONFIG_SETTINGS=y
+
+CONFIG_REBOOT=y
```

Comment:
lots of settings here are no longer needed. After the last upmerge OpenThread enabling have been simplified.

Reply:
What is the easiest way to get the list of current switches?
Can I use [this ](https://github.com/nrfconnect/sdk-nrf/blob/master/samples/openthread/cli/prj.conf)as a reference?

Reply:
the samples were not yet updated. Please use the Kconfig file for the yes/no options: https://github.com/nrfconnect/sdk-zephyr/blob/master/subsys/net/l2/openthread/Kconfig

If NET_L2_OPENTHREAD states select, the option is automatically enabled and can not be turned off so no point in setting it y.
If NET_L2_OPENTHREAD states impy, the option is default "y" so no point in setting it to "y" as well, however it can still be set to "n" by the user.
When it comes to numeric/string value it is not that simple. You need to find the definition and have a look at the defaults. Eg. MBEDTLE_HEAP_SIZE should have default 10240 now if commissioner or joiner is enabled so no point in setting it manually unless higher value is needed.

I am updating doc and samples as part of my current task, but this may take some time.

Reply:
I believe that in a case when the concurrent work is in progress, considering the scope of changes, one person should do not bother what are changes done by the other. Just to not duplicate the work needed to be done. The order of your PR's will be a determinant of who will eventually align the final changes.

Reply:
Not quite. The work is already in, what is not is the existing samples adjustments. This sample was creating by copying another that was not aligned to the new approach yet, but the "API" was already in.

Reply:
Like you said samples are not aligned yet. So I believe there will be task to do that for all of them, right?